lib/portage/manifest.py

```diff
@@ -477,7 +477,7 @@ def addFile(self, ftype, fname, hashdict=None, ignoreMissing=False):
         if hashdict is not None:
             self.fhashdict[ftype][fname].update(hashdict)
         if self.required_hashes.difference(set(self.fhashdict[ftype][fname])):
-            self.updateFileHashes(
+            self.updateAllFileHashes(
                 ftype, fname, checkExisting=False, ignoreMissing=ignoreMissing
```

Comment:
There is one more thing in [gentoo bug #875860](https://bugs.gentoo.org/875860) - fname should be in square brackets. 

At least that's what our internal testing with Petr Bahula seems like.

Reply:
Thanks - and thanks for giving me motivation to actually write a test for it, we clearly need it!